orbit/pkg/setup_experience/setup_experience.go

```diff
@@ -0,0 +1,27 @@
+package setupexperience
+
+import (
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/rs/zerolog/log"
+)
+
+// SetupExperiencer is the type that manages the Fleet setup experience flow during macOS Setup
+// Assistant. It uses swiftDialog as a UI for showing the status of software installations and
+// script execution that are configured to run before the user has full access to the device.
+type SetupExperiencer struct{}
+
+func NewSetupExperiencer() *SetupExperiencer {
+	return &SetupExperiencer{}
+}
+
+func (s *SetupExperiencer) Run(oc *fleet.OrbitConfig) error {
+	// We should only launch swiftDialog if we get the notification from Fleet.
+	if !oc.Notifications.RunSetupExperience {
```

Comment:
I'm not sure I understand why both the `installer` runner and this runner have the notification check? I would think that this is the only one that needs it (along with the check if swift dialog is installed and ready to run), but I may very well be misunderstanding.